File: Data Structures & Algorithms/trapping-rain-water/submission-1.cs
public class Solution {
    public int Trap(int[] height) {
  	int left = 0, right = height.Length - 1;
	int maxLeft = height[left], maxRight = height[right];
	int sumHeight = 0;
	int currentBlock = left;

	while (left < right)
	{
		int waterBlock = Math.Min(maxLeft, maxRight) - height[currentBlock];
		if (waterBlock > 0) sumHeight += waterBlock;

		if (height[left] <= height[right])
		{
			left++;
			currentBlock = left;
			maxLeft = Math.Max(maxLeft, height[left]);
		}
		else
		{
			right--;
			currentBlock = right;
			maxRight = Math.Max(maxRight, height[right]);

		}
	}
	return sumHeight;      
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left = 0
        right = 0
        max_length = 0
        char_set = set()
        while right < len(s):
            if s[right] not in char_set:
                char_set.add(s[right])
                right += 1
                max_length = max(max_length, right - left)
            else:
                char_set.remove(s[left])
                left += 1
            print(
                f"left: {left}, right: {right}, max_length: {max_length}, char_set: {char_set}"
            )

        return max_length